File: backend_db/app/models/invitacion.py
from sqlalchemy import Column, String, Boolean, ForeignKey, DateTime, text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from app.database import Base
from uuid import uuid4

class Invitacion(Base):
    __tablename__ = "invitaciones"

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=text("uuid_generate_v4()"))
    empresa_id = Column(UUID(as_uuid=True), ForeignKey("empresas.id", ondelete="CASCADE"), nullable=False)
    email = Column(String(150), nullable=False)
    token = Column(UUID(as_uuid=True), unique=True, nullable=False, default=uuid4)
    usada = Column(Boolean, default=False)
    fecha_creacion = Column(DateTime, server_default=func.now())
    enviada = Column(Boolean, default=False)  

File: frontend/lib/models/invitacion.dart
class Invitacion {
  final String id;
  final String empresaId;
  final String email;
  final String token;
  final bool usada;
  final DateTime fechaCreacion;

  Invitacion({
    required this.id,
    required this.empresaId,
    required this.email,
    required this.token,
    required this.usada,
    required this.fechaCreacion,
  });

  factory Invitacion.fromJson(Map<String, dynamic> json) {
    return Invitacion(
      id: json['id'],
      empresaId: json['empresa_id'],
      email: json['email'],
      token: json['token'],
      usada: json['usada'] ?? false,
      fechaCreacion: DateTime.parse(json['fecha_creacion']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'empresa_id': empresaId,
      'email': email,
      'token': token,
      'usada': usada,
      'fecha_creacion': fechaCreacion.toIso8601String(),
    };
  }
}
